adr/0017-decouple-transformation-library-from-zarf.md

```diff
@@ -0,0 +1,22 @@
+# NUMBER. TITLE
+
+Date: 2023-05-28
+
+## Status
+
+TBD
+
+## Context
+
+We need a way to decouple transformations from Zarf. We know the Zarf transformation library is battle tested, hardened, and reliable. As Pepr takes admission responsibilities from the Zarf Agent, we need a way to synchronize the TypeScript with the Go code instead of maintaining two disparate libraries which will be expected to grow.
```

Comment:
I think the context here should describe the three options a bit more:

1. WASM
2. gRPC
3. REST
4. Rewrite the code in Typescript

with simple pros and cons.

(then decision and consequences can be updated in the implementation PR)

Reply:
addressed in https://github.com/defenseunicorns/zarf/pull/1900